src/inspect_ai/model/_call_tools.py

```diff
@@ -215,9 +215,7 @@ async def call_tool_task(
                     tg.start_soon(call_tool_task, call, send_stream)
                     event._set_cancel_fn(tg.cancel_scope.cancel)
                     async with receive_stream:
-                        async for result in receive_stream:
-                            tool_message, result_event = result
-                            break
+                        tool_message, result_event = await receive_stream.receive()
```

Comment:
I think `.receive()` is the more idiomatic way of doing what you were doing. Plus, the way it was before, the stream could have closed without ever sending a value - leading to unbound `tool_message` and `result_event`.